independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/VertxResteasyReactiveRequestContext.java

```diff
@@ -60,6 +61,8 @@ public VertxResteasyReactiveRequestContext(Deployment deployment, ProvidersImpl
         context.addHeadersEndHandler(this);
         String expect = request.getHeader(HttpHeaderNames.EXPECT);
         ContextInternal internal = ((ConnectionBase) context.request().connection()).getContext();
+        ContextInternal current = (ContextInternal) Vertx.currentContext();
+        internal.localContextData().putAll(current.localContextData());
```

Comment:
@geoand @Sanne I wanted you to be aware of this change as it could have consequences on RESTEasy Reactive performances.

Reply:
Interesting, thanks for the heads up!

Question for @cescoffier: Do we always need to do this, or is this something we perhaps can do conditionally, if for example an extension needs it?

Reply:
Thanks .. yes that doesn't look very efficient. I'll add it to my list of things to check for the next round of performance work.

Reply:
For additional context: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/VertxResteasyReactiveRequestContext.

We could probably only copy the context if OTel is available (for now, it seems that it is the only case where the local context is required).

Reply:
Exactly. This should be controlled by a flag that is set only if some extension needs this capability.

I'll come up with something

Reply:
https://github.com/quarkusio/quarkus/pull/22766 is what I had in mind